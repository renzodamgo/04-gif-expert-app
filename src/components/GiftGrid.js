import React, { useEffect, useState } from 'react'
import { GifGridItem } from './GifGridItem'

export const GiftGrid = ({ category }) => {
    const [images, setImages] = useState([])

    useEffect(() => {
        getGifts()
    }, []) // solo se ejecuta una vez

    const getGifts = async () => {
        const trimCategory = category.replace(/\s/g, '+')
        const url = `https://api.giphy.com/v1/gifs/search?q=${trimCategory}&limit=5&api_key=cHrzYubYZ4nTPGmAmQrGm1aDRtL1Vlr5`
        console.log(url)
        const resp = await fetch(url);
        const { data } = await resp.json();
        const gifs = data.map(img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images.downsized_medium.url
            }
        })
        console.log(gifs)
        setImages(gifs)
    }

    // getGifts()
    return (
        <>
            <h3> {category}</h3>
            <div className='card-grid'>

                {
                    images.map((img) =>
                        <GifGridItem
                            key={img.id}
                            {...img}
                        />
                    )
                }



            </div>

        </>
    )
}
