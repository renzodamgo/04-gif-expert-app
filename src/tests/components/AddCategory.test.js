import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('Pruebas en <Add Category>/', () => {

    const setCategories = jest.fn()
    let wrapper = shallow(<AddCategory setCategories={setCategories} />);

    // Para borrar todas las iteracciones después de cada prueba
    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);

    })

    test('Debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('debe de cambiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'Hola mundo'
        input.simulate('change', { target: { value } })
        const text = wrapper.find('p').text().trim()
        expect(text).toBe(value)
    })

    test('No debe de postear información on submit', () => {
        wrapper.find('form').simulate('submit', { preventDefault() { } });

        expect(setCategories).not.toHaveBeenCalled();
    })

    test('Debe llamar el setCategories y limipiar la caja de texto', () => {
        const input = wrapper.find('input')
        const value = 'Test input'
        input.simulate('change', { target: { value } })
        wrapper.find('form').simulate('submit', { preventDefault() { } });
        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledTimes(1);
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));

        const newInput = wrapper.find('input').prop('value')
        expect(newInput).toBe('')



    })
})
