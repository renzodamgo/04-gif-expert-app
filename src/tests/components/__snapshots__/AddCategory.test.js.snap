// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Add Category>/ Debe mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    placeholder="Ingresa tu bÃºsqueda"
    type="text"
    value=""
  />
</form>
`;
